timemachine/cpp/src/neighborlist.cu

```diff
@@ -113,80 +119,131 @@ Neighborlist<RealType>::get_nblist_host(int N, const double *h_coords, const dou
     gpuErrchk(
         cudaMemcpy(&h_ixn_atoms[0], d_ixn_atoms_, MAX_ATOM_BUFFER * sizeof(unsigned int), cudaMemcpyDeviceToHost));
 
-    std::vector<std::vector<int>> ixn_list(B, std::vector<int>());
-
+    std::vector<std::vector<int>> ixn_list(row_blocks, std::vector<int>());
     for (int i = 0; i < h_ixn_count; i++) {
         int tile_idx = h_ixn_tiles[i];
-        for (int j = 0; j < 32; j++) {
-            int atom_j_idx = h_ixn_atoms[i * 32 + j];
+        for (int j = 0; j < tpb; j++) {
+            int atom_j_idx = h_ixn_atoms[i * tpb + j];
             if (atom_j_idx < N) {
                 ixn_list[tile_idx].push_back(atom_j_idx);
             }
         }
     }
-
-    gpuErrchk(cudaFree(d_coords));
+    gpuErrchk(cudaFree(d_col_coords));
+    if (K > 0) {
```

Comment:
nit: would it make sense to give a name to `K > 0` (e.g. `upper_triangular`), since it's used as a branching condition in several places? I think this might improve readability (I'm still unsure what `K > 0` means at this stage)